GLOSSARY.md

```diff
@@ -0,0 +1,284 @@
+the parent promise's resolved value or rejected error. Represents a new
+`Promise`. A handler which returns a value `T` or `Promise<T>` will resolve the
+new promise with `T`. A handler which throws an error `R`, or returns a
+`Promise` which itself eventually rejects (or has already rejected) with `R`,
+will reject the new promise with `R`.
+
+```js
+
+const p = new Promise(() => {})
+
+const p2 = p.then(function SuccessHandler(value) {
+  // fired if p resolves,
+}, function ErrorHandler(err) {
+  // fired if p rejects.
+})
+
+const p3 = p2.catch(function ErrorHandler(value) {
+  // fired if p2 rejects.
+})
+
+const p4 = p3.then(function SuccessHandler (value) { 
+  // resolved with p2's value if p2 resolves,
+  // resolved with p3's value if p2 rejects and p3 resolves
+})
+```
+
+### Settling
+
+The transition of a Promise from pending to a settled state, either "rejected"
+or "resolved." Once settled, a promise may never return to "pending" state.
+
+#### Rejection / reject
+
+The act of settling a promise with an error. Can be performed by the Executor
+or Handler `H` by throwing while `H` is on stack, or by resolving with a
+Promise that rejects at some point in its lifetime. In the case of Executors,
+rejection may additionally be achieved by calling the `reject` callback
+provided as an argument.
+
+#### Resolving / resolve
+
+The act of settling a promise with a value. Can be performed by the Executor or
+a Handler. Executors may only resolve by calling the `resolve` callback
+provided to them as an argument. In the case of Handlers, resolution is
+achieved by exiting the function normally _or_ by returning a value.
+
+Resolving a Promise `A` using a Promise `B` will cause `A` to resolve or reject
+to the same value or error as `B`.
+
+### Synchronous Rejection
+
+Sychronous rejection refers to the rare case where an executor is immediately
+rejected using `reject(<rejection>)` or `throw <rejection>`. In the case of the
+proposed Node Promise API, only invalid API use will trigger a synchronous
+rejection.
+
+### Unhandled Rejection / Rejection Handled
+
+Reported by the VM using `OnUnhandledRejection` and `OnRejectionHandled`. Called
+when a promise with no children is rejected. If the rejected promise is handled
+by a later operation, the `OnRejectionHandled` callback is called. A promise
+may synchronously transition from unhandled to handled:
+
+```js
+const p1 = new Promise(() => { throw new Error('xyz')})
+const p2 = p1.catch(() => {})
+// calls executor
+//   rejects
+//   calls unhandled rejection with p1 and error
+// calls p2.catch
+//   calls rejection handled with p1
+```
+
+Node currently keeps track of extant rejections, and fires
+`process.on('unhandledRejection')` for any rejections that have not been
+handled at the end of the microtask queue.
+
+### Microtask Queue
+
+The Microtask Queue is a specification-mandated queue of functions to be run
```

Comment:
The JS specification speaks in terms of a Job Queue and enqueuing jobs. As far as I'm aware there is no reference in the specification to the term Microtask. 

The DOM Specification speaks in terminologies like microtasks and tasks.